src/dbnode/persist/fs/msgpack/decoder.go

```diff
@@ -476,38 +481,101 @@ func (dec *Decoder) decodeIndexEntry(bytesPool pool.BytesPool) schema.IndexEntry
 	return indexEntry
 }
 
+// TODO: investigate the regular deccodeIndexEntry path to see if a similar
+// optimization where only the ID is read intially is worthwhile.
```

Comment:
As discussed, let's punt on optimizations and just call `decodeIndexEntry` here? There'll likely be some macro ones that will matter more in the short term and keeping this simple is probably ideal until we have some profiles of what's slow in the hot path.